### PR TITLE
Display selected plugin

### PR DESCRIPTION
**Pull request by kanak8278**
image

While the plugin-box styling and selection is working. The plugin comes at the top of editor panel. Trying to do the correct positioning

### CODE REVIEW

**Comment by kanak8278** [comment]
This PR is corresponding to Issue #45

**Comment by kanak8278** [comment]
image
* Moved the plugins to the DevBotFooter.

image
* Removing plugins by clicking cancel on the box, will not delete the plugin from the text box.

image
* You can add multiple plugins.
* You can't add the same plugin multiple times in the same instruction.

**Comment by sameersegal** [comment]
So now the UI is updated but the plugin information travels through chat message as well?

**Comment by kanak8278** [comment]
Yeah, Plugin information travel through text. Sure will work on it.

File: studio/src/components/Chat/DevBot.tsx [hunk 1]
```diff
@@ -16,8 +16,10 @@ interface props {
     setOnlineState: Function,
     pluginStoreToggle: Function,
     refreshIR: Function,
-    setProgramState: Function
-    inputText: string
+    setProgramState: Function,
+    inputText: string,
+    selectedPlugins: any[], // Add this prop
+    setSelectedPlugins: React.Dispatch<React.SetStateAction<any[]>> // Add this prop
```
Comment (sameersegal) [inline studio/src/components/Chat/DevBot.tsx:22]: Shouldn't this be called Function

File: studio/src/components/Chat/DevBotFooter.tsx [hunk 2]
```diff
@@ -1,14 +1,16 @@
...
+import '../../styles/selectedPlugin.css';
 
 interface props {
     disableSend: boolean
     sendMessageToWss: Function
     onFileChange: any
     pluginStoreToggle: Function
     inputText: string
+    selectedPlugins: any[]
+    setSelectedPlugins: React.Dispatch<React.SetStateAction<any[]>>
```
Comment (sameersegal) [inline studio/src/components/Chat/DevBotFooter.tsx:13]: Same comment here?

**Comment by kanak8278** [comment]
> So now the UI is updated but the plugin information travels through chat message as well?
> 
> I am going to accept this. In the next change, change the JSON structure across all levels (Frontend, Server, Engine, NL2DSL?) so that the links don't have to travel via the ChatMessage and it doesn't pollute the user's instructions.

Added it as a new issue #58

File: studio/src/components/Chat/DevBotFooter.tsx [hunk 3]
```diff
@@ -1,14 +1,16 @@
...
-
+import '../../styles/selectedPlugin.css';
 
 interface props {
     disableSend: boolean
     sendMessageToWss: Function
     onFileChange: any
     pluginStoreToggle: Function
     inputText: string
+    selectedPlugins: any[]
```
Comment (sameersegal) [inline studio/src/components/Chat/DevBotFooter.tsx:12]: Is there a way to make the data type into a `set`? Then you don't need to worry about checking whether it was added earlier or not

File: studio/src/components/PluginList/index.tsx [hunk 4]
```diff
@@ -5,13 +5,12 @@ const PluginCard = ({ plugin, pluginCallback }:any) => {
     const [expanded, setExpanded] = React.useState(true);
     const { palette } = getTheme();
 
-    const copyToClipboard = (e: Event, text:string) => {
+    const copyToClipboard = (e: React.MouseEvent, text: string, plugin: any) => {
```
Comment (sameersegal) [inline studio/src/components/PluginList/index.tsx:8]: We should drop the copyToClipboard. That was a hack. Now that we are adding it to the devFooterChatBox

**Review by sameersegal — CHANGES_REQUESTED** [review]

File: studio/src/components/PluginList/index.tsx (hunk 4 above)
Comment (kanak8278) [reply studio/src/components/PluginList/index.tsx:8]: We can't drop this. This is being used to add the url to textbox.

File: studio/src/components/Chat/DevBotFooter.tsx (hunk 3 above)
Comment (kanak8278) [reply studio/src/components/Chat/DevBotFooter.tsx]: Done

**Pull request merged**